tests/src/rules/no-unlocalized-strings.test.ts

```diff
@@ -482,6 +635,38 @@ ruleTester.run<string, Option[]>(name, rule, {
     }`,
       errors: [{ messageId: 'default' }, { messageId: 'default' }],
     },
+    {
+      name: 'object literal properties should still be checked',
+      code: `
+        const props = {
+          label: 'This should be translated'
+        };
+      `,
+      errors: [{ messageId: 'default' }],
+    },
+    {
+      name: 'regular string assignments should still be checked',
+      code: `
+        let message = 'This should be translated';
+      `,
+      errors: [{ messageId: 'default' }],
+    },
```

Comment:
Possible the duplication of the cases, we already have these: 

```js
    { code: 'const a = `Hello ${nice}`', errors },
    { code: 'export const a = `hello string`;', errors },
    { code: "const ge = 'Select tax code'", errors },
    { code: 'const a = `Foo`;', errors },

    { code: 'var a = {foo: "Bar"};', errors },
    { code: `const test = { text: 'This is not localized' }`, errors },
```

I know the test suite for this rule is a bit mess, let's not make it even more messier ) Either delete old duplicating cases, or not add a new ones. 

Also i tried to somehow group / sort them, so it's easier to spot group of functionality. Like all variants of prop assignments together. All variants of JSX text together, etc. 

Thanks!